packages/cadl-ranch-specs/http/models/bodycomplex/cadl-output/cadl/testserver/bodycomplex/_model_base.py

```diff
@@ -0,0 +1,634 @@
+
+
+class Model(_MyMutableMapping):
+    _is_model = True
+
+    def __init__(self, *args, **kwargs):
+        class_name = self.__class__.__name__
+        if len(args) > 1:
+            raise TypeError(f"{class_name}.__init__() takes 2 positional arguments but {len(args) + 1} were given")
+        dict_to_pass = {
+            rest_field._rest_name: rest_field._default
+            for rest_field in self._attr_to_rest_field.values()
+            if rest_field._default is not _UNSET
+        }
+        if args:
+            dict_to_pass.update(
+                {k: _create_value(_get_rest_field(self._attr_to_rest_field, k), v) for k, v in args[0].items()}
+            )
+        else:
+            non_attr_kwargs = [k for k in kwargs if k not in self._attr_to_rest_field]
+            if non_attr_kwargs:
+                # actual type errors only throw the first wrong keyword arg they see, so following that.
+                raise TypeError(f"{class_name}.__init__() got an unexpected keyword argument '{non_attr_kwargs[0]}'")
+            dict_to_pass.update({self._attr_to_rest_field[k]._rest_name: _serialize(v) for k, v in kwargs.items()})
+        super().__init__(dict_to_pass)
+
+    def copy(self):
+        return Model(self.__dict__)
+
+    def __new__(cls, *args: typing.Any, **kwargs: typing.Any):
+        # we know the last three classes in mro are going to be 'Model', 'dict', and 'object'
+        mros = cls.__mro__[:-3][::-1]  # ignore model, dict, and object parents, and reverse the mro order
+        attr_to_rest_field: typing.Dict[str, _RestField] = {  # map attribute name to rest_field property
+            k: v for mro_class in mros for k, v in mro_class.__dict__.items() if k[0] != "_" and hasattr(v, "_type")
+        }
+        annotations = {k: v for mro_class in mros for k, v in mro_class.__annotations__.items()}
+        for attr, rest_field in attr_to_rest_field.items():
+            rest_field._module = cls.__module__
+            if not rest_field._type:
+                rest_field._type = rest_field._get_deserialize_callable_from_annotation(annotations.get(attr, None))
+            if not rest_field._rest_name_input:
+                rest_field._rest_name_input = attr
+        cls._attr_to_rest_field: typing.Dict[str, _RestField] = {k: v for k, v in attr_to_rest_field.items()}
+
+        return super().__new__(cls)
+
+    def __init_subclass__(cls, /, discriminator=None):
+        for base in cls.__bases__:
+            if hasattr(base, "__mapping__"):
+                base.__mapping__[discriminator or cls.__name__] = cls
+
+    @classmethod
+    def _get_discriminator(cls) -> typing.Optional[str]:
+        for v in cls.__dict__.values():
+            if isinstance(v, _RestField) and v._is_discriminator:
+                return v._rest_name
+        return None
+
+    @classmethod
+    def _deserialize(cls, data):
+        if not hasattr(cls, "__mapping__"):
+            return cls(data)
```

Comment:
it seems that the check is surplus since `Model` bases on `_MyMutableMapping` which must have attribute `__mapping__`